app/views/shared/state_benefits/_add_other_state_benefits.html.erb

```diff
@@ -0,0 +1,36 @@
+<%= form_with(
+      url: form_path,
+      model: @form,
+      method: :patch,
+      local: true,
+    ) do |form| %>
+  <%= page_template page_title:, form:, template: :basic do %>
+    <span class="govuk-caption-l"><%= caption %></span>
+    <h1 class="govuk-heading-xl"><%= t(".existing", count: pluralize(@transactions.count, "benefit").to_s) %></h1>
+    <%= govuk_summary_list(classes: "summary_list_action_width_auto") do |summary_list| %>
+      <% @transactions.order(:created_at).each_with_index do |transaction, index| %>
+        <%= summary_list.with_row(html_attributes: { id: "transaction_#{index}" }) do |row| %>
+          <%= row.with_key { transaction.description } %>
+          <%= row.with_value { "#{gds_number_to_currency(transaction.amount)} #{t("transaction_types.frequencies.#{transaction.frequency}").downcase}" } %>
+          <%= row.with_action(
+                text: t("generic.change"),
+                href: url_helper.send("providers_legal_aid_application_#{url_part}_state_benefit_path", @legal_aid_application, transaction),
```

Comment:
i find the explicit passing of the existing/default url helpers odd, but can see why you need it for the `send`.

I wonder if you can just pass url another param like `show_url: providers_legal_aid_application_means_state_benefit_path || providers_legal_aid_application_partners_state_benefit_path` instead
of `url_part` 

And use
```
href: show_url(@legal_aid_application, transaction)
```